controllers/openstackmachine_controller.go

```diff
@@ -426,6 +434,36 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 	}
 	conditions.MarkTrue(openStackMachine, infrav1.APIServerIngressReadyCondition)
 
+	// Apply floating ip on worker if enabled in openStackCluster spec
+	if !util.IsControlPlaneMachine(machine) && openStackCluster.Spec.WorkerFloatingIPConfig.Enabled {
+		scope.Logger().Info("Processing worker floating IPs")
+		for _, floatingIP := range openStackCluster.Spec.WorkerFloatingIPConfig.IPAddresses {
+			fp, err := networkingService.GetOrCreateFloatingIP(openStackMachine, openStackCluster, clusterName, floatingIP)
```

Comment:
If FIP actually got created, where do we remove it on cluster deinstallation?

Reply:
For the use case where you have to create IPs first to get the actual IP address strings to put in the spec, they would need to be removed manually anyway. It would be better to use a getter instead of a get-or-create in this case.